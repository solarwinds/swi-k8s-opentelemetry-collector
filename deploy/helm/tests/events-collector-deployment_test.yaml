# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for events-collector-deployment
templates:
  - events-collector-deployment.yaml
  - events-collector-config-map.yaml
  - common-env-config-map.yaml
tests:
  - it: Events collector spec should match snapshot when using default values
    template: events-collector-deployment.yaml
    asserts:
      - matchSnapshot:
          path: spec.template.spec