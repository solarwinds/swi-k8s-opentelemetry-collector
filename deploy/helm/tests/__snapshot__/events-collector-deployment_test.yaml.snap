Events collector spec should match snapshot when using default values:
  1: |
    containers:
      - command:
          - /swi-otelcol
          - --config=/conf/relay.yaml
        env:
          - name: MY_POD_IP
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: status.podIP
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
        image: solarwinds/swi-opentelemetry-collector:0.8.6
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: 13133
        name: swi-opentelemetry-collector
        ports:
          - containerPort: 8888
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: 13133
        resources:
          limits:
            memory: 1000Mi
          requests:
            memory: 1000Mi
        securityContext: {}
        volumeMounts:
          - mountPath: /conf
            name: opentelemetry-collector-configmap
            readOnly: true
    nodeSelector:
      kubernetes.io/arch: amd64
      kubernetes.io/os: linux
    securityContext: {}
    serviceAccountName: RELEASE-NAME-swo-k8s-collector
    volumes:
      - configMap:
          items:
            - key: events.config
              path: relay.yaml
          name: RELEASE-NAME-swo-k8s-collector-events-config
        name: opentelemetry-collector-configmap
